nemo/collections/tts/inference/pipeline.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import torch
+
+from nemo.collections.tts.inference.inference_modules import (
+    AudioProcessor,
+    GraphemeToPhoneme,
+    SpectrogramSynthesizer,
+    TextProcessor,
+    TextTokenizer,
+    Vocoder,
+)
+from nemo.utils.decorators import experimental
+
+
+@experimental
+class TTSPipeline:
```

Comment:
Maybe we should make a clear name for this class and distinguish e2e from cascaded ones. If you remember, we now have both two-stage like TTS pipeline (text -> mel-gen -> audio) and an end-to-end pipeline (text -> audio).